src/main/kotlin/no/skatteetaten/aurora/boober/facade/DeployFacade.kt

```diff
@@ -103,6 +99,25 @@ class DeployFacade(
         }
     }
 
+    private fun createNamespaces(validContexts: List<AuroraDeploymentContext>): Map<String, AuroraEnvironmentResult> {
+
+        fun List<AuroraDeploymentContext>.extractNamespacesWithSampleSpec() = this
+            .groupBy { it.spec.namespace }
+            .map { (namespace, adcList) -> namespace to adcList.first().spec }
+
+        val namespacesAndSampleSpecs = validContexts.extractNamespacesWithSampleSpec()
+
+        // TODO: To reduce the amount of refactoring required to complete the AOS-5564 feature, the EnvironmentFeature
```

Comment:
Kom gjerne med kommentarer på denne TODOen. Er det ok å stoppe her, eller må/bør vi refaktorisere "i mål"?

Reply:
Jeg heller nok mot å refaktorisere, men jeg er med på at vi kan begrense scope nå for å komme i mål. Da synes jeg at det bør være en oppgave på å fikse det til f.eks neste plattform sprint.

Reply:
Er det slik at vi mister validering for denne featuren nå? Det er vel litt uheldig. Burde kanskje sånn egentlig splitte ut det du trenger i en EnvironmentService. Vi måtte da beholdt environmentfeature med shallow validering. Full validering bør vel inkorporeres i lagingen av namespaces

Reply:
Jeg må innrømme at jeg ikke tenkte på at vi nå mister gruppevalideringen. Det må inn igjen, ja. Da refaktoriserer jeg dette litt...